openedx/api.py

```diff
@@ -931,10 +930,10 @@ def enroll_in_edx_course_runs(
     edx_client = get_edx_api_service_client()
 
     if not user.openedx_users.exists() or not user.edx_username:
-        msg = (
-            f"User {user} has no Open edX user (or no username): '{user.edx_username}'"
-        )
-        raise OpenEdxUserMissingError(msg)
+        from openedx.api import reconcile_edx_username
+
+        if reconcile_edx_username(user):
+            user.refresh_from_db()
```

Comment:
This code was actually recently removed in https://github.com/mitodl/mitxonline/pull/2893 because it was causing errors with the b2b enrollment code, so bringing it back here will cause a regression with that. The consensus at the time was that whatever code was calling `enroll_in_edx_course_runs` needed to independently ensure that `openedx.api.create_user` (as oppossed to `reconcile_edx_username`) should be called by whatever code was calling `enroll_in_edx_course_runs`, ensuring it isn't called from within a transaction.

Reply:
Thank you, we have a couple of options to resolve this current issue.

- Call `reconcile_edx_username` in the [enrollment API](https://github.com/mitodl/mitxonline/blob/main/courses/views/v1/__init__.py#L330) before calling `create_run_enrollments`. This will fix the issue for the enrollments.
- Call `reconcile_edx_username` in the `create_edx_enrollments`. This has multiple uses, such as commands and old APIs.
- Catch the current OpenEdxUserMissingError exception and ask the user to fill out the profile form.

What do you suggest?

@pdpinch what are your thoughts on this?